universal-application-tool-0.0.1/app/forms/QuestionForm.java

```diff
@@ -16,19 +19,24 @@
   private Optional<Long> enumeratorId;
   private String questionText;
   private String questionHelpText;
+  private String questionExportState;
 
   protected QuestionForm() {
     questionName = "";
     questionDescription = "";
     enumeratorId = Optional.empty();
     questionText = "";
     questionHelpText = "";
+    questionExportState = "";
   }
 
   protected QuestionForm(QuestionDefinition qd) {
     questionName = qd.getName();
     questionDescription = qd.getDescription();
     enumeratorId = qd.getEnumeratorId();
+    Question q = new Question(qd);
+    q.refresh();
+    this.setQuestionExportStateFromTags(q.getQuestionTags());
```

Comment:
suggestion (if-minor): it's generally an anti-pattern to do any kind of IO or even non-trivial computation in the constructor. Could this be done elsewhere without a lot of effort?

Reply:
It would be a hassle - I'd either need to store the question definition - which is not done in any form I see - or update every question definition and require this work on every one going forward.

Reply:
Ah I wound up just storing the question definition.  /shrug seems fine